netcore/System.Private.CoreLib/src/System.Threading/Thread.cs

```diff
@@ -22,6 +22,7 @@ partial class Thread
 		IntPtr native_handle; // used only on Win32
 		/* accessed only from unmanaged code */
 		private IntPtr name;
+		private IntPtr name_generation;
```

Comment:
`int` would probably suffice. Cycle counters take a few seconds to hit 4GB, this would be 10 or 100x that.